core/src/test/java/io/questdb/griffin/LatestByTest.java

```diff
@@ -0,0 +1,235 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.std.Chars;
+import io.questdb.std.Files;
+import io.questdb.std.FilesFacadeImpl;
+import io.questdb.std.str.LPSZ;
+import org.junit.Test;
+
+public class LatestByTest extends AbstractGriffinTest {
+    @Test
+    public void testLatestByDoesNotNeedFullScan() throws Exception {
+        assertMemoryLeak(() -> {
+            ff = new FilesFacadeImpl() {
+                @Override
+                public long openRO(LPSZ name) {
+                    // Query should not scan the first partition
+                    // all the latest values are in the second partition
+                    if (Chars.contains(name, "1970-01-01")) {
+                        return -1;
+                    }
+                    return Files.openRO(name);
+                }
+            };
+            compile("create table t as (" +
+                    "select rnd_symbol('a', 'b') s, timestamp_sequence(0, 60*60*1000*1000L) ts from long_sequence(49)" +
+                    ") timestamp(ts) Partition by DAY");
+
+            assertQuery("s\tts\n" +
+                            "a\t1970-01-02T23:00:00.000000Z\n" +
+                            "b\t1970-01-03T00:00:00.000000Z\n",
+                    "t " +
+                            "where s in ('a', 'b') " +
+                            "latest on ts partition by s",
+                    "ts",
+                    true,
+                    true);
+        });
+    }
+
```

Comment:
Could you change test name or add comment to make it clear that query doesn't need to scan any partition because symbol sets in where clause and table don't overlap ?

Reply:
done

Reply:
The comment I see is 
// Query should not scan the first partition
// all the latest values are in the second partition
which is not correct because actually cursor doesn't scan table at all . 
I think name testLatestBySymbolEmpty() is not clear enough .